lib/resty/kafka/producer.lua

```diff
@@ -293,25 +293,22 @@ end
 
 
 _flush_buffer = function (self)
+    if self._flush_pending_flag then
```

Comment:
the `_flush_pending_flag` variable name is not meaningful.

Reply:
and I think the logical for this `flag` is wrong, I think we should use `_flush_buffer_pending` here.

and setting `_flush_buffer_pending` in this way
````
local ok, err = timer_at(0, _flush, self)
if ok then
    self._flush_buffer_pending = true
    return
end
-- XX: should log the error message here.
ngx.log(...)
```